sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -211,6 +213,34 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
                     }
                 });
 
+            // Enable custom events and set filter to enable collection.
+            builder.Services.AddSingleton<IApplicationInsightsEventLogger, ApplicationInsightsEventLogger>();
+
+            // The default behavior is to always capture logs for custom events.
+            // This can achieved with this code level filter -> loggingBuilder.AddFilter<OpenTelemetry.Logs.ApplicationInsightsLoggerProvider>("",LogLevel.Information);
+            // However, this may run into issues when users try to override this behavior from Configuration like below using appsettings.json:
+            // {
+            //   "Logging": {
+            //     "OpenTelemetry": {
+            //       "LogLevel": {
+            //         "Azure.Monitor.OpenTelemetry.CustomEvents": "None"
+            //       }
+            //     }
+            //   },
+            //   ...
+            // }
+            // The reason is as both rules will match the filter, the last one added wins.
+            // To ensure that the default filter is in the beginning of filter rules, so that user override from Configuration will always win,
+            // we add code filter rule to the 0th position as below.
+            builder.Services.Configure<LoggerFilterOptions>(
+                options => options.Rules.Insert(
+                    0,
+                    new LoggerFilterRule(
+                        "OpenTelemetry.Logs.OpenTelemetryLoggerProvider",
```

Comment:
Just peaked at the docs for `LoggerFilterRule` it seems you can specify [provider type or alias](https://learn.microsoft.com/dotnet/api/microsoft.extensions.logging.loggerfilterrule.providername). The alias we use for OTel is "OpenTelemetry" (defined [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/1b3f1894d95fd01fe475245ec60ef273d5f77bb6/src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggerProvider.cs#L16)) should we do that here instead? Just thinking the alias would be nice to not tie this code to the implementation SDK happens to be using.

Reply:
+1

Reply:
updated